packages/@lwc/engine-server/src/types.ts

```diff
@@ -61,11 +62,11 @@ export interface HostElement {
     // explicitly given only a getter, so it doesn't need to be a Symbol.
     tagName: string;
     [HostNamespaceKey]: string;
-    [HostParentKey]: HostElement | null;
+    [HostParentKey]: HostElement | HostShadowRoot | null;
```

Comment:
I'm not sure why the host parent can be a host shadow root, whatever that means.